### PR TITLE
Use webpack

### PR DESCRIPTION
**Pull request by francoiscampbell**
## Description

- Uses webpack to bundle and deploy
- The changelog is mandatory but dummy for now

### DevQA Prep

- Please `npm install`, this PR uses a new package

### DevQA Steps
- `make build` should build `yvm.js` and copy `yvm.sh` to `artifacts/webpack_build`
- `make build_and_deploy` should build and also zip up the files into `artifacts/webpack_build/yvm.zip`, and attempt to create a release on github from the dummy changelog (it will fail because of no auth)
- Go to https://github.com/settings/tokens, create a token, and run `GITHUB_TOKEN=<token> make build_and_deploy` (now it will pass)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -51,10 +51,14 @@ yvm-test:
 
 # ---- Infrastructure for Test/Deploy ----
 
-# TODO: finish the babel stuff
 .PHONY: build
 build: node_modules
-	babel $(BABEL_FLAGS)
+	@node_modules/.bin/webpack
```
Comment (jakebolam) [inline Makefile]: [nb] can just do webpack

File: scripts/install.sh [hunk 2]
```diff
@@ -19,11 +20,11 @@ fi
 mkdir -p ${install_dir}
 
 if [ "$use_local" = true ]; then
-    cp -R * ${install_dir}
+    cp "${artifacts_dir}/yvm.sh" "${artifacts_dir}/yvm.js" ${install_dir}
 else
-    curl -o ${zip_install_path} ${release_url}
-    unzip ${zip_install_path} -d ${install_dir}
-    rm ${zip_install_path}
+    curl -o ${zip_download_path} ${release_url}
```
Comment (jakebolam) [inline scripts/install.sh]: How is release_url set?

File: webpack/webpack.config.deploy.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+const baseConfig = require('./webpack.config.base')
+const GithubReleasePlugin = require('./githubReleasePlugin')
+const ZipPlugin = require('zip-webpack-plugin')
+
+const zipfileName = 'yvm.js'
+baseConfig.plugins.push(
+    new ZipPlugin({
+        filename: zipfileName,
+    }),
+    new GithubReleasePlugin({
```
Comment (jakebolam) [inline webpack/webpack.config.deploy.js]: This is cool :party parrot

File: webpack/githubReleasePlugin.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+const ghRelease = require('gh-release')
+
+module.exports = class GithubReleasePlugin {
+    constructor(options) {
```
Comment (jakebolam) [inline webpack/githubReleasePlugin.js]: v v cool

File: .circleci/config.yml [hunk 5]
```diff
@@ -1,24 +1,50 @@
 version: 2
+
+docker_defaults: &docker_defaults
```
Comment (jakebolam) [inline .circleci/config.yml]: Also cool

File: .circleci/config.yml [hunk 6]
```diff
@@ -1,24 +1,50 @@
...
-      - restore_cache:
-          name: Restore node_modules cache
-          key: yvm-{{ .Branch }}-{{ checksum "package-lock.json" }}
-      - run: make node_modules
-      - save_cache:
-          name: Save node_modules cache
-          key: yvm-{{ .Branch }}-{{ checksum "package-lock.json" }}
-          paths:
-            - node_modules/
+      - *restore_cache
```
Comment (jakebolam) [inline .circleci/config.yml]: Do we not need to build/save NPM packages here too?

File: scripts/install.sh (hunk 2 above)
Comment (francoiscampbell) [reply scripts/install.sh]: it'll be a direct url to the latest master zip, it's just not implemented yet

File: webpack/webpack.config.deploy.js (hunk 3 above)
Comment (francoiscampbell) [reply webpack/webpack.config.deploy.js]: yeah, i was thinking of making a separate npm package too (not now, maybe later)

File: .circleci/config.yml (hunk 6 above)
Comment (francoiscampbell) [reply .circleci/config.yml]: for the tests? no, jest runs on the source files, not the built bundle

**Comment by umar-tophat** [comment]
## ✅ QA
- running `make build` creates bundle and copies yvm.sh file
- running `GITHUB_TOKEN=<token> make build_and_deploy` creates zip file containing necessary files and attempts to push release to GitHub (wasn't able to actually complete the process because a release already exists)

**Pull request merged**